projects/Canary/Models/Record.cs

```diff
@@ -172,28 +195,51 @@ public string FhirInfo
             }
         }
 
-        public Record()
+        public Record(bool retrieveFsh = false)
         {
             record = this.CreateEmptyRecord();
             this.ije = this.CreateIJEFromRecord(this.record, false).ToString();
+            if (retrieveFsh) {
+                System.Threading.Tasks.Task<string> task =
+                            System.Threading.Tasks.Task.Run<string>(async () => await Record.GetFshData(this.record.ToJson()));
```

Comment:
What are some of the special considerations to asynchronous processing?

Reply:
The user would expect to await for the response.  Each of the API calls are blocking calls in the pre-existing code.

Reply:
Would using synchronous calls be more appropriate here? To block further processing until results are returned.

Reply:
That's actually what's happening with the await keyword.  It's blocked until the results return.

Reply:
My understanding is that the `await` keyword makes non-blocking/async calls, which is the exact opposite than what you described.

Reply:
It's blocked until it returns, we can go through the debugger if you like.